commons-numbers-complex/src/main/java/org/apache/commons/numbers/complex/ComplexFunctions.java

```diff
@@ -677,6 +929,48 @@ private static <R> R log(DoubleUnaryOperator log,
         return action.apply(re, arg(real, imaginary));
     }
 
+    /**
+     * Returns the complex power of the first complex number raised to the power of
+     * second complex number.
+     * Implements the formula:
+     *
+     * <p>\[ z^x = e^{x \ln(z)} \]
+     *
+     * <p>If the complex number is zero then this method returns zero if the second complex number is positive
+     * in the real component and zero in the imaginary component;
+     * otherwise it returns NaN + iNaN.
+     *
+     * @param re1 Real part \( a \) of the first complex number \( (a +ib) \).
+     * @param im1 Imaginary part \( b \) of the first complex number \( (a +ib) \).
+     * @param re2 Real part \( a \) of the second complex number \( (a +ib) \).
+     * @param im2 Imaginary part \( b \) of the second complex number \( (a +ib) \).
+     * @param action Consumer for the power of the complex number.
+     * @param <R> the return type of the supplied action.
+     * @return the object returned by the supplied action.
+     * @see #log(double, double, ComplexSink)
+     * @see #multiply(double, double, double, double, ComplexSink)
+     * @see #exp(double, double, ComplexSink)
+     * @see <a href="http://mathworld.wolfram.com/ComplexExponentiation.html">Complex exponentiation</a>
+     * @see <a href="http://functions.wolfram.com/ElementaryFunctions/Power/">Power</a>
+     */
+    public static <R> R pow(double re1, double im1, double re2, double im2, ComplexSink<R> action) {
+        if (re1 == 0 &&
+            im1 == 0) {
+            // This value is zero. Test the other.
+            if (re2 > 0 &&
+                im2 == 0) {
+                // 0 raised to positive number is 0
+                return action.apply(0, 0);
+            }
+            // 0 raised to anything else is NaN
+            return action.apply(Double.NaN, Double.NaN);
+        }
+        final ComplexUnaryOperator<R> log = ComplexFunctions::log;
```

Comment:
These declarations could be done in a static final variable. However we do not need to to use the `andThen` composition:
```Java
        return log(re1, im1, (x, y) -> {
            return multiply(x, y, re2, im2, (a, b) -> {
                return exp(a, b, action);
            });
        });
```
I am undecided which is preferred. However to avoid adding the `andThen` function just to support this composition I would use the explicit lambda code above.